rfq/manager.go

```diff
@@ -965,6 +966,33 @@ func (m *Manager) GetAssetGroupKey(ctx context.Context,
 	return groupKeyBytes, nil
 }
 
+// AssetMatchesSpecifier checks if the provided asset satisfies the provided
+// specifier. If the specifier includes a group key, we will check if the asset
+// belongs to that group.
+func (m *Manager) AssetMatchesSpecifier(ctx context.Context,
+	specifier asset.Specifier, id asset.ID) (bool, error) {
+
+	switch {
+	case specifier.HasGroupPubKey():
+		group, err := m.GetAssetGroupKey(ctx, id)
+		if err != nil {
+			return false, err
+		}
+
+		specifierGK := specifier.UnwrapGroupKeyToPtr()
+
+		return bytes.Equal(group, specifierGK.SerializeCompressed()),
+			nil
+
+	case specifier.HasId():
+		specifierID := specifier.UnwrapIdToPtr()
+
+		return specifierID.IsEqual(id), nil
+	}
+
+	return false, fmt.Errorf("specifier is empty")
```

Comment:
style nit: I find it easier to read if a `switch` has a `default` case that has the error return. But personal preference I guess.